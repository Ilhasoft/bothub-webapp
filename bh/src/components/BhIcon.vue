<template>
  <span
    :class="{
      'bh-icon': true,
      [`bh-icon--${size}`]: !!size,
      'bh-icon--spin': spin,
    }"
    @click="$emit('click', $event)"
    v-html="svg"
  />
</template>

<script>
import icons from '../icons';


export default {
  name: 'BhIcon',
  props: {
    value: {
      type: String,
      default: 'help-text',
    },
    size: {
      type: String,
      default: null,
    },
    spin: {
      type: Boolean,
      default: false,
    },
  },
  computed: {
    svg() {
      return icons[this.value];
    },
  },
};
</script>

<style lang="scss">
@import '@scss/variables.scss';


@keyframes BhIconSpin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

.bh {
  &-icon {
    display: inline-block;
    width: 1em;
    height: 1em;
    vertical-align: middle;

    svg {
      display: block;
      width: 100%;
      height: 100%;
      fill: currentColor;
    }

    &--spin {
      svg {
        animation-name: BhIconSpin;
        animation-duration: 1s;
        animation-timing-function: linear;
        animation-iteration-count: infinite;
      }
    }

    @each $size in $sizes {
      $size-name: nth($size, 1);
      $size-value: nth($size, 2);

      &--#{$size-name} {
        width: $size-value;
        height: $size-value;
      }
    }
  }
}
</style>
