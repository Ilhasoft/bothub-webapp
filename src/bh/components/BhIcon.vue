<template>
  <span
    :class="{
      'bh-icon': true,
      [`bh-icon--${size}`]: !!size,
    }"
    @click="$emit('click', $event)"
    v-html="svg" />
</template>

<script>
import { icons } from '../utils';


export default {
  name: 'BhIcon',
  props: {
    value: {
      type: String,
      default: 'help-text',
    },
    size: {
      type: String,
      default: null,
    },
  },
  computed: {
    svg() {
      return icons[this.value];
    },
  },
};
</script>

<style lang="scss">
.bh {
  &-icon {
    display: inline-block;
    width: 1.25rem;
    height: 1.25rem;
    padding: 0 !important;
    vertical-align: text-top;

    svg {
      display: block;
      width: 100%;
      max-width: 100%;
      height: 100%;
      max-height: 100%;
      fill: currentColor;
    }

    &--small {
      width: 1rem;
      height: 1rem;
    }

    &--medium {
      width: 1.5rem;
      height: 1.5rem;
    }

    &--large {
      width: 1.75rem;
      height: 1.75rem;
    }
  }
}
</style>
