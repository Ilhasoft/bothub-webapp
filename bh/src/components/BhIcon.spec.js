import { shallowMount, createLocalVue } from '@vue/test-utils';

import BH from '..';
import BhIcon from './BhIcon';


const localVue = createLocalVue();
localVue.use(BH);

describe('BhIcon.vue', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallowMount(BhIcon, {
      localVue,
    });
  });

  test('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
