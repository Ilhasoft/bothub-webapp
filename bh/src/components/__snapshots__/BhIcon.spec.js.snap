// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BhIcon.vue renders correctly 1`] = `
VueWrapper {
  "_emitted": Object {},
  "_emittedByOrder": Array [],
  "isFunctionalComponent": undefined,
}
`;
